components/reddit/reddit.app.js

```diff
@@ -0,0 +1,189 @@
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return (await axios(opts)).data;
+    },
+    wereThingsPulled(reddit_things) {
+      const things = get(reddit_things, "data.children");
+      return things && things.length;
+    },
+    did4xxErrorOccurred(err) {
+      return (
+        get(err, "response.status") !== undefined &&
+        get(err, "response.status") !== null &&
+        err.response.status >= 400
+      );
+    },
+    async getNewHotSubredditPosts(subreddit, g, show, sr_detail, limit = 100) {
+      const params = {};
+      if (show) {
+        params["show"] = "all";
+      }
+      params["g"] = g;
+      params["sr_detail"] = sr_detail;
+      params["limit"] = limit;
+      return await this._makeRequest({
+        path: `/r/${subreddit}/hot`,
+        params,
+      });
+    },
+    async getNewSubredditLinks(before_link, subreddit, limit = 100) {
+      return await this._makeRequest({
+        path: `/r/${subreddit}/new`,
+        params: {
+          before: before_link,
+          limit,
+        },
+      });
+    },
+    async getNewSubredditComments(
+      subreddit,
+      subreddit_id36,
+      context,
+      depth,
+      sr_detail,
+      limit = 100
+    ) {
+      const resp = await this._makeRequest({
+        path: `/r/${subreddit}/comments/article`,
+        params: {
+          article: subreddit_id36,
+          context,
+          depth,
+          limit,
+          sort: "new",
+          sr_detail,
+          theme: "default",
+          threaded: true,
+          trucate: 0,
+        },
+      });
+      return resp[1];
+    },
+    async getNewUserLinks(
+      before,
+      username,
+      context,
+      t,
+      sr_detail,
+      limit = 100
+    ) {
+      const params = {
+        before,
+        context,
+        show: "given",
+        sort: "new",
+        t,
+        type: "links",
+        sr_detail,
+        limit,
+      };
+
+      return await this._makeRequest({
+        path: `/user/${username}/submitted`,
+        params,
+      });
+    },
+    async getNewUserComments(
+      before,
+      username,
+      context,
+      t,
+      sr_detail,
+      limit = 100
+    ) {
+      const params = {
+        before,
+        context,
+        show: "given",
+        sort: "new",
+        t,
+        type: "comments",
+        sr_detail,
+        limit,
+      };
+      return await this._makeRequest({
+        path: `/user/${username}/comments`,
+        params,
+      });
+    },
+    async searchSubreddits(after_link, query) {
+      const params = {
+        after: after_link,
+        limit: 100,
+        q: query,
+        show_users: false,
+        sort: "relevance",
+        sr_detail: false,
+        typeahead_active: false,
+      };
+      return await this._makeRequest({
+        path: `/subreddits/search`,
+        params,
+      });
+    },
+    async getAllSearchSubredditsResults(query) {
+      var after = null;
```

Comment:
As a convention, we use `let` to declare mutable variables.

Reply:
Ok, I will prefer let onwards.